### ISSUE TITLE
Reimplement existing features on Floem Lapce

### ISSUE THREAD

**Issue by MinusGix** [lapce/lapce#2417]
The Floem (UI library) version of Lapce will be merged into the `master` branch soon. Once that occurs, bringing it completely up to feature parity with the druid version of Lapce is needed. There is a variety of missing features, some of which were skipped over and some of which have happened since the fork, that need to be reimplemented.  
  
**Easy**: Features that are just a matter of porting code over
[ ] Completion Lens
[ ] IME phantom text
  - might be blocked by backend?
**Medium:** May need a good amount of changes and work
?  
**Hard:** May be intricate to implement, even with the existing implementation. Either due to backend differences
[ ] Diff Editor
  - Should be separate from the document instead.

**Comment by TitanNano** [comment]
Wayland support is also no longer a thing.

**Comment by WeetHet** [comment]
And basically nothing works on macos anymore

**Comment by MinusGix** [comment]
> And basically nothing works on macos anymore

Can you be more specific? I believe dzhou uses the editor on macos, so editing works (etc). Is there a specific feature that's missing? A specific integration with MacOS?

**Comment by WeetHet** [comment]
> > And basically nothing works on macos anymore
> 
> Can you be more specific? I believe dzhou uses the editor on macos, so editing works (etc). Is there a specific feature that's missing? A specific integration with MacOS?

1. No macOS menu bar integration, meaning no way to open a folder/create new file as
2. Shortcuts don't seem to work
3. LSP also is not working at all somehow
4. Closing last window closes the app — not macOS way of doing things
5. And probably a lot more stuff I didn't found in the first thirty seconds of using nightly build

**Comment by MinusGix** [comment]
Thanks. I'll add those to the list and discuss them.

> 3. LSP also is not working at all somehow

Which specific LSP?

**Comment by WeetHet** [comment]
> Thanks. I'll add those to the list and discuss them.
> 
> > 3. LSP also is not working at all somehow
> 
> Which specific LSP?

Rust

**Comment by dsweb19778** [comment]
![Screenshot_20230604_214730](https://github.com/lapce/lapce/assets/40462602/95fea256-657f-449a-a48c-ed93e56cc844)
after updating to the new nightly version all is broken nothing working setting dont open shortcuts also 
Os : Linux Manjaro Kde

**Comment by panekj** [comment]
> ![Screenshot_20230604_214730](https://user-images.githubusercontent.com/40462602/243206453-95fea256-657f-449a-a48c-ed93e56cc844.png) after updating to the new nightly version all is broken nothing working setting dont open shortcuts also Os : Linux Manjaro Kde

that's expected, use stable release if you want working editor

**Comment by dsweb19778** [comment]
am using nightly version many weeks now and was fine just last update break all

**Comment by panekj** [comment]
> am using nightly version many weeks now and was fine just last update break all

as I said, that is expected, we're rewriting most of the codebase, use stable release if you want working editor

**Comment by tk103331** [comment]
Are there only these tasks to handle? There are still many differences when I run the floem version.

**Comment by panekj** [comment]
> Are there only these tasks to handle? There are still many differences when I run the floem version.

These are the only written down. If something is missing, it should be noted here.

**Comment by Nyabinary** [comment]
Diff editor is reimplemented now, isn't it?

**Comment by MinusGix** [comment]
> Diff editor is reimplemented now, isn't it?

Yep. I just forgot to mark it. Thanks.

**Comment by moon989** [comment]
期待新版本，融合 **floem UI** 框架

**Comment by 1268** [comment]
https://github.com/lapce/lapce/issues/2550

**Comment by WeetHet** [comment]
On macOS (and linux probably?) the keybindings are weird for colemak. Instead of using real keystrokes, lapce seems to use "places" on the keyboard. For example to press Cmd+N and create a new file, one should press Cmd+K, as the position of K in Colemak coincides with position of N in QWERTY. This should be changed, so lapce would use real key inputs and not their places

**Comment by WeetHet** [comment]
Also #2801

**Comment by WeetHet** [comment]
@MinusGix mind adding this two to the list? Also menu bar support on macOS seems to be long gone again
